groups/bsl/bslh/bslh_spookyhashalgorithm.h

```diff
@@ -380,7 +380,7 @@ SpookyHashAlgorithm::SpookyHashAlgorithm()
 inline
 SpookyHashAlgorithm::SpookyHashAlgorithm(const char *seed)
 : d_state(
-#if !defined(BSLS_PLATFORM_CPU_X86_64) && !defined(BSLS_PLATFORM_CPU_X86)
+#if !defined(BSLS_PLATFORM_IS_LITTLE_ENDIAN)
```

Comment:
This should also not be checking for little_endian - this is about whether the cpu handles unaligned reads properly.  For now i would leave this as-is and we can engage the optimization at a later date if we make the m1 a more fully supported platform.